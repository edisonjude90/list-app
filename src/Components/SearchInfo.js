import React from "react";
import { connect } from "react-redux";

const mapStateToProps = (state) => ({
    viewResult:state.viewDetails
});

const ConnectSearchInfo = ({viewResult})=>{
return (
        <div className="search-info">
            {(viewResult.first) &&
                <div>
                    <header><h2>Info</h2></header>
                    <div>
                        <label> First Name : { viewResult.first } </label>
                    </div>
                    <div>
                        <label> Last Name : { viewResult.last } </label>
                    </div>
                    <div>
                        <label> Age : { viewResult.age } </label>
                    </div>
                    <div>
                        <label> Date Of Birth : { viewResult.dob } </label>
                    </div>
                    <div>
                        <label> Location : { viewResult.location } </label>
                    </div>
                    <div>
                        <label> Description : { viewResult.description } </label>
                    </div>
                </div>
            }
        </div>
    
)};


const SearchInfo = connect(mapStateToProps)(ConnectSearchInfo);

export default SearchInfo;