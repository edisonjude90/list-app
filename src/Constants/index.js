export const SEARCH_PERSON = "SEARCH_PERSON";

export const VIEW_PERSON_DETAILS = "VIEW_PERSON_DETAILS";