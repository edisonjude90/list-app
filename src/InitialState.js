const AppState = {
    personList:[
        {
            id: 1,
            first: "John",
            last: "Roberts",
            age: 51,
            dob: "24-05-1967",
            location: "Chicago, IL",
            description: "John is a retired YouTuber"
        },
        {
            id: 2,
            first: "Frita",
            last: "Waters",
            age: 27,
            dob: "24-05-1991",
            location: "Phoenix, AZ",
            description: "Frita loves the desert, cheese, and turtles."
        },
        {
            id: 3,
            first: "Bart",
            last: "Simpson",
            age: 24,
            dob: "24-05-1996",
            location: "Springfield, IL",
            description: "Bart skateboards everywhere."
        },
        {
            id: 4,
            first: "Harry",
            last: "Simpson",
            age: 24,
            dob: "24-05-1996",
            location: "New York, US",
            description: "Likes Nature"
        },
        {
            id: 5,
            first: "David",
            last: "Mark",
            age: 24,
            dob: "24-05-1990",
            location: "Phoenix, AZ",
            description: "David is a DJ"
        },
        {
            id: 6,
            first: "Jason",
            last: "Williams",
            age: 28,
            dob: "10-01-1990",
            location: "New York, US",
            description: "Likes Bowling"
        }
    ],
    searchResult:[],
    viewDetails:{}
};

export default AppState;