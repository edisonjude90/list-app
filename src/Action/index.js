import { SEARCH_PERSON, VIEW_PERSON_DETAILS } from "../Constants";

export const SEARCH_PERSON_ACTION = (payload) => ({
    type:SEARCH_PERSON,
    payload:payload
});

export const VIEW_PERSON_DETAILS_ACTION = (payload) => ({
    type:VIEW_PERSON_DETAILS,
    payload:payload
});
