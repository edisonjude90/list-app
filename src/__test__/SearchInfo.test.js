import React from 'react';
import { createStore } from "redux";
import renderer from 'react-test-renderer';

import InitialState from '../InitialState';
import { VIEW_PERSON_DETAILS } from "../Constants";
import { VIEW_PERSON_DETAILS_ACTION } from "../Action";
import reducer from '../Reducer';

import SearchInfo from '../Components/SearchInfo';

it('SearchInfo Component Displays Users Info', ()=>{
    const store = createStore(reducer,InitialState);
    store.dispatch(VIEW_PERSON_DETAILS_ACTION({id:1}));
    const SearchInfoSection = renderer.create(<SearchInfo store = { store } />).toJSON();
    expect(SearchInfoSection).toMatchSnapshot();
});
