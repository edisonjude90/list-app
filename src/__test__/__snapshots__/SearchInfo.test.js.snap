// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInfo Component Displays Users Info 1`] = `
<div
  className="search-info"
>
  <div>
    <header>
      <h2>
        Info
      </h2>
    </header>
    <div>
      <label>
         First Name : 
        John
         
      </label>
    </div>
    <div>
      <label>
         Last Name : 
        Roberts
         
      </label>
    </div>
    <div>
      <label>
         Age : 
        51
         
      </label>
    </div>
    <div>
      <label>
         Date Of Birth : 
        24-05-1967
         
      </label>
    </div>
    <div>
      <label>
         Location : 
        Chicago, IL
         
      </label>
    </div>
    <div>
      <label>
         Description : 
        John is a retired YouTuber
         
      </label>
    </div>
  </div>
</div>
`;
